actors/user/src/main/java/org/sunbird/user/actors/UserProfileReadActor.java

```diff
@@ -106,11 +108,43 @@ private void getUserProfile(Request actorMessage) {
 
   private Response getUserProfileData(Request actorMessage) {
     Map<String, Object> userMap = actorMessage.getRequest();
+    String id = (String) userMap.get(JsonKey.USER_ID);
+    String userId;
+    String provider = (String) actorMessage.getContext().get(JsonKey.PROVIDER);
+    String idType = (String) actorMessage.getContext().get(JsonKey.ID_TYPE);
+    boolean showMaskedData = false;
+    if (!StringUtils.isEmpty(provider)) {
+      if (StringUtils.isEmpty(idType)) {
+        userId = null;
+        ProjectCommonException.throwClientErrorException(
+            ResponseCode.mandatoryParamsMissing,
+            MessageFormat.format(
+                ResponseCode.mandatoryParamsMissing.getErrorMessage(), JsonKey.ID_TYPE));
+      } else {
+        HashMap<String, Object> map = new HashMap<>();
+        map.put(JsonKey.EXTERNAL_ID, id);
+        map.put(JsonKey.EXTERNAL_ID_PROVIDER, provider);
+        map.put(JsonKey.EXTERNAL_ID_TYPE, idType);
+        userId = UserUtil.getUserIdFromExternalId(map);
```

Comment:
Use an existing method (in UserExternalIdentityDao) to find user by external ID from Cassandra (rather than Elasticsearch which involves sync delay) as this API is used in SSO user creation where we need to ensure that no two users are created with same external ID.